sdk/containerregistry/Azure.Containers.ContainerRegistry/tests/ContainerRegistryClientLiveTests.cs

```diff
@@ -42,7 +43,62 @@ await foreach (string repository in repositories)
                 }
             }
 
+            // Assert
             Assert.IsTrue(gotHelloWorld);
         }
+
+        [RecordedTest]
+        public async Task CanGetRepositoriesWithCustomPageSize()
+        {
+            // Arrange
+            var client = CreateClient();
+            int pageSize = 2;
+            int minExpectedPages = 2;
+
+            // Act
+            AsyncPageable<string> repositories = client.GetRepositoriesAsync();
+            var pages = repositories.AsPages(pageSizeHint: pageSize);
+
+            int pageCount = 0;
+            await foreach (var page in pages)
+            {
+                Assert.IsTrue(page.Values.Count <= pageSize);
```

Comment:
Should we be asserting equality? If pageSizeHint wasn't specified would the test still pass right now?

Reply:
My thinking was that if we have multiple pages, the last one could come back with fewer elements than would be in a full page, depending on the number of resources stored in the test service instance (which I don't want to rely on in the test).

If pageSizeHint wasn't specified, the number of elements in a page wouldn't be limited, so it could be many more than the requested page size.